sources/Interop/D3D12MemoryAllocator/src/AllocationObjectAllocator.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace TerraFX.Interop
+{
+    /// <summary>Thread-safe wrapper over PoolAllocator free list, for allocation of Allocation objects.</summary>
+    internal unsafe struct AllocationObjectAllocator
+    {
+        private D3D12MA_MUTEX m_Mutex;
+        private PoolAllocator<Allocation> m_Allocator;
+
+        public AllocationObjectAllocator(ALLOCATION_CALLBACKS* allocationCallbacks)
+        {
+            D3D12MA_MUTEX.Init(out m_Mutex);
+            m_Allocator = new PoolAllocator<Allocation>(allocationCallbacks, 1024);
+        }
+
+        public Allocation* Allocate(AllocatorPimpl* allocator, ulong size, int wasZeroInitialized)
+        {
+            using MutexLock mutexLock = new((D3D12MA_MUTEX*)Unsafe.AsPointer(ref m_Mutex));
```

Comment:
I really don't like the `new()` code, I find it much more readable using `var` on the LHS and the type on the RHS
```suggestion
            using var mutexLock = new MutexLock((D3D12MA_MUTEX*)Unsafe.AsPointer(ref m_Mutex));
```

Reply:
Done in 7ee6bda6f831499895afae74b9d241fd98c1d613, though I cried a little 😢